src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -393,6 +393,14 @@ private Object readResolve() {
             if (tagsToPush == null) {
                 this.pushMerge = true;
             }
+            this.configVersion = 1L;
+        }
+
+        if (this.configVersion < 2L && isPushBranches()) {
+            for (BranchToPush branch : branchesToPush){
+                 branch.setSrcBranchName("HEAD");
+            }
+            this.configVersion = 2L;
```

Comment:
I not sure wether this change is needed because jenkins doesn't save job changes after load that can have readResolve() updates.